optuna/study.py

```diff
@@ -237,10 +237,9 @@ def optimize(
                 The number of parallel jobs. If this argument is set to :obj:`-1`, the number is
                 set to CPU counts.
             catch:
-                A study continues to run even when a trial raises one of exceptions specified in
-                this argument. Default is (`Exception <https://docs.python.org/3/library/
-                exceptions.html#Exception>`_,), where all non-exit exceptions are handled
-                by this logic.
+                A study continues to run even when a trial raises one of the exceptions specified
+                in this argument. Default is an empty tuple, i.e. the study will stop for any
+                exception except for :class:`structs.TrialPruned`.
```

Comment:
I think that Optuna usually doesn't use the absolute path to refer to a class:
```suggestion
                exception except for :class:`~structs.TrialPruned`.
```